src/dotnet/ToolPackage/IToolPackageFactory.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal interface IToolPackageFactory
+    {
+        (IToolPackageStore, IToolPackageInstaller) CreateToolPackageStoreAndInstaller(
```

Comment:
I think we may need to separate this out into two different factories, unfortunately.  I'm just thinking about the future when the uninstall command accepts a `--tool-path` option.  It doesn't need an installer.

Reply:
The problem is ToolPackageInstaller need ToolPackageStore object in constructor. So ToolPackageInstaller factory need to take a ToolPackageStore as input, it is even more odd.

In uninstall, i do need to have a separate CreateToolPackageStore method.

Reply:
I'm :+1: with just adding a `CreateToolPackageStore` method to this type for the uninstaller in the future.